Placeholder for "Speed up SELinux volume relabeling using mounts" alpha feature (#35667)
kubernetes/website: content/en/docs/tasks/configure-pod-container/security-context.md

To assign SELinux labels, the SELinux security module must be loaded on the host operating system.
{{< /note >}}

> ### Speeding up SELinux volume relabeling

Reviewer: ```suggestion
### Efficient SELinux volume relabeling
```
“speeding up” might not make as much sense once the feature is stable. We like to write [timeless](https://developers.google.com/style/timeless-documentation) documentation if we can.

Author: Great suggestion! Applied

{{< feature-state for_k8s_version="v1.25" state="alpha" >}}

By default, the contrainer runtime recursively assigns SELinux label to all
files on all Pod volumes. To speed up this process, Kubernetes can change the
SELinux label of a volume instantly by using a mount option
`-o context=<label>`.

To benefit from this speedup, all these conditions must be met:

> * Alpha feature gates `ReadWriteOncePod` and `SELinuxMountReadWriteOncePod` must

Reviewer 2: Are these gates documented?

Author: Added `SELinuxMountReadWriteOncePod` to docs/reference/command-line-tools-reference/feature-gates.md.
`ReadWriteOncePod` is already there.

Reviewer: Thank.

  be enabled.

> * Pod must use PersistentVolumeClaim with `accessModes` `ReadWriteOncePod`.

Reviewer 2: ```suggestion
* Pod must use PersistentVolumeClaim with `accessModes: ReadWriteOncePod`.
```

Author: It's an array, so updated to `accessModes: [ReadWriteOncePod]`

Reviewer: Ah OK. I'd write: `accessModes: ["ReadWriteOncePod"]`  though - readers can immediately then see that it's a list of strings.

Author: Fixed

* Pod (or all its Containers that use the PersistentVolumeClaim) must
  have `seLinuxOptions` set.
* Corresponding PersistentVolume must be either a CSI volume or an in-tree iSCSI

>   volume.

Author: It is.

  * For the CSI volume, the corresponding CSI driver must announce that it
    supports mounting with `-o context` by setting `spec.seLinuxMount: true` in
    its CSIDriver instance.

All other volumes will be relabeled by the container runtime by recursively
changing SELinux label of all files on the volumes.

{{< note >}}

> In Kubernetes 1.25, kubelet looses track of volume labels after restart. In

Reviewer 3: typo: looses -> loses

Author: Fixed

> In Kubernetes 1.25, kubelet loses track of volume labels after restart. In

Reviewer: Also, clarify what is restarted (I assume you mean that the kubelet or the whole node has restarted, but it's not 100% clear).

> other words, kubelet may refuse to start Pods with errors like "conflicting

Reviewer: ```suggestion
other words, then kubelet may refuse to start Pods with errors similar to  "conflicting
```

> other words, kubelet may to refuse to start Pods with errors like "conflicting

Reviewer 3: may to refuse -> may refuse

Author: Fixed

> SELinux labels of volume", while there are no conflicting in Pods. Make sure

Reviewer 3: "no conflicting" -> "no conflicts" or "no conflicting labels"

Author: fixed

nodes are fully drained before restarting kubelet. This will be addressed in a

> future Kubernetes release before graduating the feature to Beta.

Reviewer: We [avoid statements about the future](https://kubernetes.io/docs/contribute/style/style-guide/#avoid-statements-about-the-future) unless these are firm commitments that SIG Architecture would stand behind.

If not fixing that bug for beta would be a release blocker, **and this is clearly defined in the KEP**, then this statement is OK to include. Even then, I'd be minded to omit it. Folks are reading docs for a Kubernetes release that has happened and usually don't need to find out how to operate Kubernetes releases we haven't yet published.

Also also: consider linking to https://kubernetes.io/docs/tasks/administer-cluster/safely-drain-node/

Author: Ack, added link to node draining + removed "This will be addressed in a future Kubernetes release before graduating the feature to Beta.".
